<!DOCTYPE html>
<html lang="en"><head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8"><meta charset="utf-8"><title data-loaded="true" data-rh="data-loaded">UniProtKB return fields | UniProt help | UniProt</title><meta name="viewport" content="width=device-width,initial-scale=1"><meta name="theme-color" content="#00639a"><link rel="apple-touch-icon" sizes="180x180" href="https://www.uniprot.org/apple-touch-icon.png"><link rel="icon" type="image/png" sizes="32x32" href="https://www.uniprot.org/favicon-32x32.png"><link rel="icon" type="image/png" sizes="16x16" href="https://www.uniprot.org/favicon-16x16.png"><link rel="manifest" href="https://www.uniprot.org/manifest.json"><link rel="preconnect" href="https://fonts.gstatic.com/"><link rel="preconnect" href="https://rest.uniprot.org/"><link rel="preload" as="style" href="https://fonts.googleapis.com/css?family=Lato:400,700|Source+Sans+Pro:600,700&amp;display=swap"><script async="" src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/google-analytics_analytics.js"></script><script>window.dataLayer = window.dataLayer || [];
      function gtag() {
        dataLayer.push(arguments);
      }
      gtag('js', new Date());

      gtag('config', 'UA-6228219-1', {
        anonymize_ip: true,
        allow_google_signals: false,
        allow_ad_personalization_signals: false,
        restricted_data_processing: true,
        custom_map: {
          dimension1: 'result_view',
          metric1: 'result_view_set',
        },
        send_page_view: false,
      });</script><script>(function (h, o, t, j, a, r) {
        h.hj =
          h.hj ||
          function () {
            (h.hj.q = h.hj.q || []).push(arguments);
          };
        h._hjSettings = { hjid: 2638207, hjsv: 6 };
        a = o.getElementsByTagName('head')[0];
        r = o.createElement('script');
        r.async = 1;
        r.src = t + h._hjSettings.hjid + j + h._hjSettings.hjsv;
        a.appendChild(r);
      })(window, document, 'https://static.hotjar.com/c/hotjar-', '.js?sv=');</script><script async="" src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/hotjar-2638207.js"></script><script defer="defer" src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/app.f595dd.js"></script><script defer="defer" src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/app.a482ef.js"></script><script defer="defer" src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/app.e98bca.js"></script><link href="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/main.53939a.css" rel="stylesheet"><link rel="stylesheet" type="text/css" href="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/help-entry.7c6835.css"><link rel="stylesheet" type="text/css" href="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/footer.2723a7.css"><meta name="description" content="UniProt is the worldâ€™s leading high-quality, comprehensive and freely accessible resource of protein sequence and functional information." data-rh="true"><script type="application/ld+json">{
  "@context": "https://schema.org",
  "@type": "WebSite",
  "url": "https://www.uniprot.org",
  "potentialAction": {
    "@type": "SearchAction",
    "target": {
      "@type": "EntryPoint",
      "urlTemplate": "https://www.uniprot.org/uniprotkb?query={q}"
    },
    "query-input": "required name=q"
  }
}</script><script type="application/ld+json">{
  "@context": "https://schema.org",
  "@type": "Organization",
  "url": "https://www.uniprot.org",
  "logo": [
    "https://www.uniprot.org/android-chrome-512x512.png",
    "/uniprot-logo.img.0df091.svg"
  ]
}</script><link rel="stylesheet" type="text/css" href="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/contextual-help.43aea7.css"><link rel="stylesheet" type="text/css" href="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/back-to-the-top.b5e880.css"></head><body><div id="root"><div class="base-layout"><header class="main-header"><div class="header"><div class="header__logo"><a href="https://www.uniprot.org/"><div class="ETRuS" aria-label="UniProt home page"></div></a></div><div class="header__navigation"><ul class="no-bullet rZ_Xt"><li><a href="https://www.uniprot.org/blast">BLAST</a></li><li><a href="https://www.uniprot.org/align">Align</a></li><li><a href="https://www.uniprot.org/peptide-search">Peptide search</a></li><li><a href="https://www.uniprot.org/id-mapping">ID mapping</a></li><li><a href="https://sparql.uniprot.org/" target="_blank" rel="noopener">SPARQL</a></li></ul></div><div class="header__search"><section role="search"><form aria-label="Main search" class="main-search"><div class="dropdown" aria-expanded="false" aria-haspopup="true"><button class="button primary" style="--main-button-color: #00639a;" type="button">UniProtKB</button></div><div class="main-search__input-container"><input type="text" aria-label="Text query in uniprotkb" style="--input-padding: 12ch;"><div class="main-search__secondary-container"><button type="button" class="main-search--secondary"><small>Advanced</small></button><small> | </small><button type="button" class="main-search--secondary"><small>List</small></button></div></div><button class="button primary" type="submit">Search</button></form></section></div><div class="header__secondary"><div class="header__navigation"><a title="Tools dashboard" class="qyi35 Ljg_Z" href="https://www.uniprot.org/tool-dashboard"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" fill="currentColor" width="1.4em"><path d="M503 215l-46-46c-6-6-14-9-22-9h-51V80c0-27-21-48-48-48H176c-27 0-48 21-48 48v80H77c-8 0-16 3-22 9L9 215c-6 6-9 14-9 22v83h128v-16c0-9 7-16 16-16h32c9 0 16 7 16 16v16h128v-16c0-9 7-16 16-16h32c9 0 16 7 16 16v16h128v-83c0-8-3-16-9-22zm-183-55H192V96h128v64zm64 208c0 9-7 16-16 16h-32c-9 0-16-7-16-16v-16H192v16c0 9-7 16-16 16h-32c-9 0-16-7-16-16v-16H0v96c0 18 14 32 32 32h448c18 0 32-14 32-32v-96H384v16z"></path></svg></a><a title="Basket" class="qyi35 Ljg_Z" href="https://www.uniprot.org/basket/uniprotkb"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512" fill="currentColor" width="1.4em"><path d="M576 216v16c0 13-11 24-24 24h-8l-26 183c-3 23-24 41-48 41H106c-24 0-45-18-48-41L32 256h-8c-13 0-24-11-24-24v-16c0-13 11-24 24-24h67L198 45a32 32 0 0152 38l-80 109h236L326 83a32 32 0 0152-38l107 147h67c13 0 24 11 24 24zM312 392V280a24 24 0 10-48 0v112a24 24 0 1048 0zm112 0V280a24 24 0 10-48 0v112a24 24 0 1048 0zm-224 0V280a24 24 0 10-48 0v112a24 24 0 1048 0z"></path></svg></a><a title="Contact" class="qyi35" href="https://www.uniprot.org/contact"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 13 10" fill="currentColor" width="1.4em"><path fill-rule="evenodd" d="M12 1.25C12 .56 11.46 0 10.8 0H1.2C.54 0 0 .56 0 1.25v7.5C0 9.44.54 10 1.2 10h9.6c.66 0 1.2-.56 1.2-1.25v-7.5zm-1.2 0L6 4.38 1.2 1.25h9.6zM6 5.63L1.2 2.5v6.25h9.6V2.5L6 5.63z"></path></svg></a><a href="https://www.uniprot.org/help">Help</a></div></div></div></header><section class="in-page-messages"><div></div></section><div class="main-content"><div class="single-column-layout"><main class="single-column-layout__main"><h1>UniProtKB return fields</h1><section class="HSbN4 card"><div class="card__container"><div class="card__content"><div><h2 id="uniprotkb-column-names-for-programmatic-access">UniProtKB column names for programmatic access</h2>
<p>This document lists the differences between the returned columns by RESTful APIs.
User can ask required columns returned by an API by passing the <em>Returned Field</em> in the request url
and the response will have those requested fields.</p>
<p>A specific page listing cross-reference fields can be found here: <a href="https://www.uniprot.org/help/return_fields_databases">Return Fields Databases</a>.</p>
<p>For example, to get UniProtKB entry with accession and proteomes in new API run the below <code>curl</code> command:</p>
<pre><code>curl https://rest.uniprot.org/uniprotkb/search?query=human&amp;fields=accession,xref_proteomes
</code></pre>
<p>Equivalent request in the current website would be :</p>
<pre><code>https://www.uniprot.org/uniprotkb?query=human&amp;columns=id,proteome
</code></pre>
<p>The <code>Label</code> is the readable name of the returned field. The <code>Label</code> is shown on the website/TSV or in Excel format.</p>
<h2 id="names--taxonomy">Names &amp; Taxonomy</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Entry</td>
<td>id</td>
<td>accession</td>
</tr>
<tr>
<td>Entry name</td>
<td>entry name</td>
<td>id</td>
</tr>
<tr>
<td>Gene names</td>
<td>genes</td>
<td>gene_names</td>
</tr>
<tr>
<td>Gene names (primary)</td>
<td>genes(PREFERRED)</td>
<td>gene_primary</td>
</tr>
<tr>
<td>Gene names (synonym)</td>
<td>genes(ALTERNATIVE)</td>
<td>gene_synonym</td>
</tr>
<tr>
<td>Gene names (ordered locus)</td>
<td>genes(OLN)</td>
<td>gene_oln</td>
</tr>
<tr>
<td>Gene names (ORF)</td>
<td>genes(ORF)</td>
<td>gene_orf</td>
</tr>
<tr>
<td>Organism</td>
<td>organism</td>
<td>organism_name</td>
</tr>
<tr>
<td>Organism ID</td>
<td>organism-id</td>
<td>organism_id</td>
</tr>
<tr>
<td>Protein names</td>
<td>protein names</td>
<td>protein_name</td>
</tr>
<tr>
<td>Proteomes</td>
<td>proteome</td>
<td>xref_proteomes</td>
</tr>
<tr>
<td>Taxonomic lineage</td>
<td>lineage(ALL)</td>
<td>lineage</td>
</tr>
<tr>
<td>Taxonomic lineage (IDs)</td>
<td>lineage(ALL)</td>
<td>lineage_ids</td>
</tr>
<tr>
<td>Virus hosts</td>
<td>virus hosts</td>
<td>virus_hosts</td>
</tr>
</tbody></table>
<p><em>* Label in the old and new API unless otherwise specified</em></p>
<h2 id="sequences">Sequences</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Alternative products</td>
<td>comment(ALTERNATIVE PRODUCTS)</td>
<td>cc_alternative_products</td>
</tr>
<tr>
<td>Alternative sequence</td>
<td>feature(ALTERNATIVE SEQUENCE)</td>
<td>ft_var_seq</td>
</tr>
<tr>
<td>Erroneous gene model prediction</td>
<td>comment(ERRONEOUS GENE MODEL PREDICTION)</td>
<td>error_gmodel_pred</td>
</tr>
<tr>
<td>Fragment</td>
<td>fragment</td>
<td>fragment</td>
</tr>
<tr>
<td>Gene encoded by</td>
<td>encodedon</td>
<td>organelle</td>
</tr>
<tr>
<td>Length</td>
<td>length</td>
<td>length</td>
</tr>
<tr>
<td>Mass</td>
<td>mass</td>
<td>mass</td>
</tr>
<tr>
<td>Mass spectrometry</td>
<td>comment(MASS SPECTROMETRY)</td>
<td>cc_mass_spectrometry</td>
</tr>
<tr>
<td>Natural variant</td>
<td>feature(NATURAL VARIANT)</td>
<td>ft_variant</td>
</tr>
<tr>
<td>Non-adjacent residues</td>
<td>feature(NON ADJACENT RESIDUES)</td>
<td>ft_non_cons</td>
</tr>
<tr>
<td>Non-standard residue</td>
<td>feature(NON STANDARD RESIDUE)</td>
<td>ft_non_std</td>
</tr>
<tr>
<td>Non-terminal residue</td>
<td>feature(NON TERMINAL RESIDUE)</td>
<td>ft_non_ter</td>
</tr>
<tr>
<td>Polymorphism</td>
<td>comment(POLYMORPHISM)</td>
<td>cc_polymorphism</td>
</tr>
<tr>
<td>RNA editing</td>
<td>comment(RNA EDITING)</td>
<td>cc_rna_editing</td>
</tr>
<tr>
<td>Sequence</td>
<td>sequence</td>
<td>sequence</td>
</tr>
<tr>
<td>Sequence caution</td>
<td>comment(SEQUENCE CAUTION)</td>
<td>cc_sequence_caution</td>
</tr>
<tr>
<td>Sequence conflict</td>
<td>feature(SEQUENCE CONFLICT)</td>
<td>ft_conflict</td>
</tr>
<tr>
<td>Sequence uncertainty</td>
<td>feature(SEQUENCE UNCERTAINTY)</td>
<td>ft_unsure</td>
</tr>
<tr>
<td>Sequence version</td>
<td>version(sequence)</td>
<td>sequence_version</td>
</tr>
</tbody></table>
<h2 id="function">Function</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Absorption</td>
<td>comment(ABSORPTION)</td>
<td>absorption</td>
</tr>
<tr>
<td>Active site</td>
<td>feature(ACTIVE SITE)</td>
<td>ft_act_site</td>
</tr>
<tr>
<td>Activity regulation</td>
<td>comment(ACTIVITY REGULATION)</td>
<td>cc_activity_regulation</td>
</tr>
<tr>
<td>Binding site</td>
<td>feature(BINDING SITE)</td>
<td>ft_binding</td>
</tr>
<tr>
<td>Calcium binding</td>
<td>chebi</td>
<td>ft_ca_bind</td>
</tr>
<tr>
<td>Catalytic activity</td>
<td>chebi(Catalytic activity)</td>
<td>cc_catalytic_activity</td>
</tr>
<tr>
<td>Cofactor</td>
<td>chebi(Cofactor)</td>
<td>cc_cofactor</td>
</tr>
<tr>
<td>DNA binding</td>
<td>feature(DNA BINDING)</td>
<td>ft_dna_bind</td>
</tr>
<tr>
<td>EC number</td>
<td>ec</td>
<td>ec</td>
</tr>
<tr>
<td>Function [CC]</td>
<td>comment(FUNCTION)</td>
<td>cc_function</td>
</tr>
<tr>
<td>Kinetics</td>
<td>comment(KINETICS)</td>
<td>kinetics</td>
</tr>
<tr>
<td>Metal binding</td>
<td>feature(METAL BINDING)</td>
<td>ft_metal</td>
</tr>
<tr>
<td>Nucleotide binding</td>
<td>feature(NP BIND)</td>
<td>ft_np_bind</td>
</tr>
<tr>
<td>Pathway</td>
<td>comment(PATHWAY)</td>
<td>cc_pathway</td>
</tr>
<tr>
<td>pH dependence</td>
<td>comment(PH DEPENDENCE)</td>
<td>ph_dependence</td>
</tr>
<tr>
<td>Redox potential</td>
<td>comment(REDOX POTENTIAL)</td>
<td>redox_potential</td>
</tr>
<tr>
<td>Rhea ID</td>
<td>rhea-id</td>
<td>rhea</td>
</tr>
<tr>
<td>Site</td>
<td>feature(SITE)</td>
<td>ft_site</td>
</tr>
<tr>
<td>Temperature dependence</td>
<td>comment(TEMPERATURE DEPENDENCE)</td>
<td>temp_dependence</td>
</tr>
</tbody></table>
<h2 id="miscellaneous">Miscellaneous</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Annotation</td>
<td>annotation score</td>
<td>annotation_score</td>
</tr>
<tr>
<td>Caution</td>
<td>comment(CAUTION)</td>
<td>cc_caution</td>
</tr>
<tr>
<td>Comment Count</td>
<td>&lt;does not exist&gt;</td>
<td>comment_count</td>
</tr>
<tr>
<td>Features</td>
<td>features</td>
<td>feature_count</td>
</tr>
<tr>
<td>Keyword ID</td>
<td>keyword-id</td>
<td>keywordid</td>
</tr>
<tr>
<td>Keywords</td>
<td>keywords</td>
<td>keyword</td>
</tr>
<tr>
<td>Matched text</td>
<td>context</td>
<td>&lt;does not exist&gt;</td>
</tr>
<tr>
<td>Miscellaneous [CC]</td>
<td>comment(MISCELLANEOUS)</td>
<td>cc_miscellaneous</td>
</tr>
<tr>
<td>Protein existence</td>
<td>existence</td>
<td>protein_existence</td>
</tr>
<tr>
<td>Reviewed</td>
<td>reviewed</td>
<td>reviewed</td>
</tr>
<tr>
<td>Tools</td>
<td>tools</td>
<td>tools</td>
</tr>
<tr>
<td>UniParc</td>
<td>uniparcid</td>
<td>uniparc_id</td>
</tr>
</tbody></table>
<h2 id="interaction">Interaction</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Interacts with</td>
<td>interactor</td>
<td>cc_interaction</td>
</tr>
<tr>
<td>Subunit structure[CC]</td>
<td>comment(SUBUNIT)</td>
<td>cc_subunit</td>
</tr>
</tbody></table>
<h2 id="expression">Expression</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Developmental stage</td>
<td>comment(DEVELOPMENTAL STAGE)</td>
<td>cc_developmental_stage</td>
</tr>
<tr>
<td>Induction</td>
<td>comment(INDUCTION)</td>
<td>cc_induction</td>
</tr>
<tr>
<td>Tissue specificity</td>
<td>comment(TISSUE SPECIFICITY)</td>
<td>cc_tissue_specificity</td>
</tr>
</tbody></table>
<h2 id="gene-ontology-go">Gene Ontology (GO)</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Gene ontology (biological process)</td>
<td>go(biological process)</td>
<td>go_p</td>
</tr>
<tr>
<td>Gene ontology (cellular component)</td>
<td>go(cellular component)</td>
<td>go_c</td>
</tr>
<tr>
<td>Gene ontology (GO)</td>
<td>go</td>
<td>go</td>
</tr>
<tr>
<td>Gene ontology (molecular function)</td>
<td>go(molecular function)</td>
<td>go_f</td>
</tr>
<tr>
<td>Gene ontology IDs</td>
<td>go-id</td>
<td>go_id</td>
</tr>
</tbody></table>
<h2 id="pathology--biotech">Pathology &amp; Biotech</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Allergenic properties</td>
<td>comment(ALLERGEN)</td>
<td>cc_allergen</td>
</tr>
<tr>
<td>Biotechnological use</td>
<td>comment(BIOTECHNOLOGY)</td>
<td>cc_biotechnology</td>
</tr>
<tr>
<td>Disruption phenotype</td>
<td>comment(DISRUPTION PHENOTYPE)</td>
<td>cc_disruption_phenotype</td>
</tr>
<tr>
<td>Involvement in disease</td>
<td>comment(DISEASE)</td>
<td>cc_disease</td>
</tr>
<tr>
<td>Mutagenesis</td>
<td>feature(MUTAGENESIS)</td>
<td>ft_mutagen</td>
</tr>
<tr>
<td>Pharmaceutical use</td>
<td>comment(PHARMACEUTICAL)</td>
<td>cc_pharmaceutical</td>
</tr>
<tr>
<td>Toxic dose</td>
<td>comment(TOXIC DOSE)</td>
<td>cc_toxic_dose</td>
</tr>
</tbody></table>
<h2 id="subcellular-location">Subcellular location</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Intramembrane</td>
<td>feature(INTRAMEMBRANE)</td>
<td>ft_intramem</td>
</tr>
<tr>
<td>Subcellular location[CC]</td>
<td>comment(SUBCELLULAR LOCATION)</td>
<td>cc_subcellular_location</td>
</tr>
<tr>
<td>Topological domain</td>
<td>feature(TOPOLOGICAL DOMAIN)</td>
<td>ft_topo_dom</td>
</tr>
<tr>
<td>Transmembrane</td>
<td>feature(TRANSMEMBRANE)</td>
<td>ft_transmem</td>
</tr>
</tbody></table>
<h2 id="ptm--processsing">PTM / Processsing</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Chain</td>
<td>feature(CHAIN)</td>
<td>ft_chain</td>
</tr>
<tr>
<td>Cross-link</td>
<td>feature(CROSS LINK)</td>
<td>ft_crosslnk</td>
</tr>
<tr>
<td>Disulfide bond</td>
<td>feature(DISULFIDE BOND)</td>
<td>ft_disulfid</td>
</tr>
<tr>
<td>Glycosylation</td>
<td>feature(GLYCOSYLATION)</td>
<td>ft_carbohyd</td>
</tr>
<tr>
<td>Initiator methionine</td>
<td>feature(INITIATOR METHIONINE)</td>
<td>ft_init_met</td>
</tr>
<tr>
<td>Lipidation</td>
<td>feature(LIPIDATION)</td>
<td>ft_lipid</td>
</tr>
<tr>
<td>Modified residue</td>
<td>feature(MODIFIED RESIDUE)</td>
<td>ft_mod_res</td>
</tr>
<tr>
<td>Peptide</td>
<td>feature(PEPTIDE)</td>
<td>ft_peptide</td>
</tr>
<tr>
<td>Post-translational modification</td>
<td>comment(PTM)</td>
<td>cc_ptm</td>
</tr>
<tr>
<td>Propeptide</td>
<td>feature(PROPEPTIDE)</td>
<td>ft_propep</td>
</tr>
<tr>
<td>Signal peptide</td>
<td>feature(SIGNAL)</td>
<td>ft_signal</td>
</tr>
<tr>
<td>Transit peptide</td>
<td>feature(TRANSIT)</td>
<td>ft_transit</td>
</tr>
</tbody></table>
<h2 id="structure">Structure</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>3D</td>
<td>3d</td>
<td>structure_3d</td>
</tr>
<tr>
<td>Beta strand</td>
<td>feature(BETA STRAND)</td>
<td>ft_strand</td>
</tr>
<tr>
<td>Helix</td>
<td>feature(HELIX)</td>
<td>ft_helix</td>
</tr>
<tr>
<td>Turn</td>
<td>feature(TURN)</td>
<td>ft_turn</td>
</tr>
</tbody></table>
<h2 id="publications">Publications</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Mapped PubMed ID</td>
<td>citationmapping</td>
<td>&lt;does not exist&gt;</td>
</tr>
<tr>
<td>PubMed ID</td>
<td>citation</td>
<td>lit_pubmed_id</td>
</tr>
</tbody></table>
<h2 id="date-of">Date of</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Date of creation</td>
<td>created</td>
<td>date_created</td>
</tr>
<tr>
<td>Date of last modification</td>
<td>last-modified</td>
<td>date_modified</td>
</tr>
<tr>
<td>Date of last sequence modification</td>
<td>sequence-modified</td>
<td>date_sequence_modified</td>
</tr>
<tr>
<td>Entry version</td>
<td>version(entry)</td>
<td>version</td>
</tr>
</tbody></table>
<h2 id="family--domains">Family &amp; Domains</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Coiled coil</td>
<td>feature(COILED COIL)</td>
<td>ft_coiled</td>
</tr>
<tr>
<td>Compositional bias</td>
<td>feature(COMPOSITIONAL BIAS)</td>
<td>ft_compbias</td>
</tr>
<tr>
<td>Domain[CC]</td>
<td>comment(DOMAIN)</td>
<td>cc_domain</td>
</tr>
<tr>
<td>Domain[FT]</td>
<td>feature(DOMAIN EXTENT)</td>
<td>ft_domain</td>
</tr>
<tr>
<td>Motif</td>
<td>feature(MOTIF)</td>
<td>ft_motif</td>
</tr>
<tr>
<td>Protein families</td>
<td>families</td>
<td>protein_families</td>
</tr>
<tr>
<td>Region</td>
<td>feature(REGION)</td>
<td>ft_region</td>
</tr>
<tr>
<td>Repeat</td>
<td>feature(REPEAT)</td>
<td>ft_repeat</td>
</tr>
<tr>
<td>Sequence similarities</td>
<td>comment(SIMILARITY)</td>
<td>&lt;does not exist&gt;</td>
</tr>
<tr>
<td>Zinc finger</td>
<td>feature(ZINC FINGER)</td>
<td>ft_zn_fing</td>
</tr>
</tbody></table>
<h2 id="sequence-databases">Sequence Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>CCDS</td>
<td>database(CCDS)</td>
<td>xref_ccds</td>
</tr>
<tr>
<td>EMBL</td>
<td>database(EMBL)</td>
<td>xref_embl</td>
</tr>
<tr>
<td>PIR</td>
<td>database(PIR)</td>
<td>xref_pir</td>
</tr>
<tr>
<td>RefSeq</td>
<td>database(RefSeq)</td>
<td>xref_refseq</td>
</tr>
</tbody></table>
<h2 id="3d-structure-databases">3D Structure Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>AlphaFoldDB</td>
<td>database(AlphaFoldDB)</td>
<td>xref_alphafolddb</td>
</tr>
<tr>
<td>BMRB</td>
<td>database(BMRB)</td>
<td>xref_bmrb</td>
</tr>
<tr>
<td>PCDDB</td>
<td>database(PCDDB)</td>
<td>xref_pcddb</td>
</tr>
<tr>
<td>PDB</td>
<td>database(PDB)</td>
<td>xref_pdb</td>
</tr>
<tr>
<td>PDBsum</td>
<td>database(PDBsum)</td>
<td>xref_pdbsum</td>
</tr>
<tr>
<td>SASBDB</td>
<td>database(SASBDB)</td>
<td>xref_sasbdb</td>
</tr>
<tr>
<td>SMR</td>
<td>database(SMR)</td>
<td>xref_smr</td>
</tr>
</tbody></table>
<h2 id="protein-protein-interaction-databases">Protein-Protein Interaction Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>BioGRID</td>
<td>database(BioGRID)</td>
<td>xref_biogrid</td>
</tr>
<tr>
<td>CORUM</td>
<td>database(CORUM)</td>
<td>xref_corum</td>
</tr>
<tr>
<td>ComplexPortal</td>
<td>database(ComplexPortal)</td>
<td>xref_complexportal</td>
</tr>
<tr>
<td>DIP</td>
<td>database(DIP)</td>
<td>xref_dip</td>
</tr>
<tr>
<td>ELM</td>
<td>database(ELM)</td>
<td>xref_elm</td>
</tr>
<tr>
<td>IntAct</td>
<td>database(IntAct)</td>
<td>xref_intact</td>
</tr>
<tr>
<td>MINT</td>
<td>database(MINT)</td>
<td>xref_mint</td>
</tr>
<tr>
<td>STRING</td>
<td>database(STRING)</td>
<td>xref_string</td>
</tr>
</tbody></table>
<h2 id="chemistry-databases">Chemistry Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>BindingDB</td>
<td>database(BindingDB)</td>
<td>xref_bindingdb</td>
</tr>
<tr>
<td>ChEMBL</td>
<td>database(ChEMBL)</td>
<td>xref_chembl</td>
</tr>
<tr>
<td>DrugBank</td>
<td>database(DrugBank)</td>
<td>xref_drugbank</td>
</tr>
<tr>
<td>DrugCentral</td>
<td>database(DrugCentral)</td>
<td>xref_drugcentral</td>
</tr>
<tr>
<td>GuidetoPHARMACOLOGY</td>
<td>database(GuidetoPHARMACOLOGY)</td>
<td>xref_guidetopharmacology</td>
</tr>
<tr>
<td>SwissLipids</td>
<td>database(SwissLipids)</td>
<td>xref_swisslipids</td>
</tr>
</tbody></table>
<h2 id="protein-familygroup-databases">Protein Family/Group Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Allergome</td>
<td>database(Allergome)</td>
<td>xref_allergome</td>
</tr>
<tr>
<td>CAZy</td>
<td>database(CAZy)</td>
<td>xref_cazy</td>
</tr>
<tr>
<td>CLAE</td>
<td>database(CLAE)</td>
<td>xref_clae</td>
</tr>
<tr>
<td>ESTHER</td>
<td>database(ESTHER)</td>
<td>xref_esther</td>
</tr>
<tr>
<td>IMGT_GENE-DB</td>
<td>database(IMGT GENE-DB)</td>
<td>xref_imgt_gene-db</td>
</tr>
<tr>
<td>MEROPS</td>
<td>database(MEROPS)</td>
<td>xref_merops</td>
</tr>
<tr>
<td>MoonDB</td>
<td>database(MoonDB)</td>
<td>xref_moondb</td>
</tr>
<tr>
<td>MoonProt</td>
<td>database(MoonProt)</td>
<td>xref_moonprot</td>
</tr>
<tr>
<td>PeroxiBase</td>
<td>database(PeroxiBase)</td>
<td>xref_peroxibase</td>
</tr>
<tr>
<td>REBASE</td>
<td>database(REBASE)</td>
<td>xref_rebase</td>
</tr>
<tr>
<td>TCDB</td>
<td>database(TCDB)</td>
<td>xref_tcdb</td>
</tr>
<tr>
<td>UniLectin</td>
<td>database(UniLectin)</td>
<td>xref_unilectin</td>
</tr>
</tbody></table>
<h2 id="ptm-databases">Ptm Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>CarbonylDB</td>
<td>database(CarbonylDB)</td>
<td>xref_carbonyldb</td>
</tr>
<tr>
<td>DEPOD</td>
<td>database(DEPOD)</td>
<td>xref_depod</td>
</tr>
<tr>
<td>GlyConnect</td>
<td>database(GlyConnect)</td>
<td>xref_glyconnect</td>
</tr>
<tr>
<td>GlyGen</td>
<td>database(GlyGen)</td>
<td>xref_glygen</td>
</tr>
<tr>
<td>MetOSite</td>
<td>database(MetOSite)</td>
<td>xref_metosite</td>
</tr>
<tr>
<td>PhosphoSitePlus</td>
<td>database(PhosphoSitePlus)</td>
<td>xref_phosphositeplus</td>
</tr>
<tr>
<td>SwissPalm</td>
<td>database(SwissPalm)</td>
<td>xref_swisspalm</td>
</tr>
<tr>
<td>UniCarbKB</td>
<td>&lt;does not exist&gt;</td>
<td>xref_unicarbkb</td>
</tr>
<tr>
<td>iPTMnet</td>
<td>database(iPTMnet)</td>
<td>xref_iptmnet</td>
</tr>
</tbody></table>
<h2 id="genetic-variation-databases">Genetic Variation Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>BioMuta</td>
<td>database(BioMuta)</td>
<td>xref_biomuta</td>
</tr>
<tr>
<td>DMDM</td>
<td>database(DMDM)</td>
<td>xref_dmdm</td>
</tr>
<tr>
<td>dbSNP</td>
<td>database(dbSNP)</td>
<td>xref_dbsnp</td>
</tr>
</tbody></table>
<h2 id="2d-gel-databases">2D Gel Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>COMPLUYEAST-2DPAGE</td>
<td>database(COMPLUYEAST-2DPAGE)</td>
<td>xref_compluyeast-2dpage</td>
</tr>
<tr>
<td>DOSAC-COBS-2DPAGE</td>
<td>database(DOSAC-COBS-2DPAGE)</td>
<td>xref_dosac-cobs-2dpage</td>
</tr>
<tr>
<td>OGP</td>
<td>database(OGP)</td>
<td>xref_ogp</td>
</tr>
<tr>
<td>REPRODUCTION-2DPAGE</td>
<td>database(REPRODUCTION-2DPAGE)</td>
<td>xref_reproduction-2dpage</td>
</tr>
<tr>
<td>SWISS-2DPAGE</td>
<td>database(SWISS-2DPAGE)</td>
<td>xref_swiss-2dpage</td>
</tr>
<tr>
<td>UCD-2DPAGE</td>
<td>database(UCD-2DPAGE)</td>
<td>xref_ucd-2dpage</td>
</tr>
<tr>
<td>World-2DPAGE</td>
<td>database(World-2DPAGE)</td>
<td>xref_world-2dpage</td>
</tr>
</tbody></table>
<h2 id="proteomic-databases">Proteomic Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>CPTAC</td>
<td>database(CPTAC)</td>
<td>xref_cptac</td>
</tr>
<tr>
<td>EPD</td>
<td>database(EPD)</td>
<td>xref_epd</td>
</tr>
<tr>
<td>MassIVE</td>
<td>database(MassIVE)</td>
<td>xref_massive</td>
</tr>
<tr>
<td>MaxQB</td>
<td>database(MaxQB)</td>
<td>xref_maxqb</td>
</tr>
<tr>
<td>PRIDE</td>
<td>database(PRIDE)</td>
<td>xref_pride</td>
</tr>
<tr>
<td>PaxDb</td>
<td>database(PaxDb)</td>
<td>xref_paxdb</td>
</tr>
<tr>
<td>PeptideAtlas</td>
<td>database(PeptideAtlas)</td>
<td>xref_peptideatlas</td>
</tr>
<tr>
<td>ProMEX</td>
<td>database(ProMEX)</td>
<td>xref_promex</td>
</tr>
<tr>
<td>ProteomicsDB</td>
<td>database(ProteomicsDB)</td>
<td>xref_proteomicsdb</td>
</tr>
<tr>
<td>TopDownProteomics</td>
<td>database(TopDownProteomics)</td>
<td>xref_topdownproteomics</td>
</tr>
<tr>
<td>jPOST</td>
<td>database(jPOST)</td>
<td>xref_jpost</td>
</tr>
</tbody></table>
<h2 id="protocols-and-materials-databases">Protocols And Materials Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>ABCD</td>
<td>database(ABCD)</td>
<td>xref_abcd</td>
</tr>
<tr>
<td>Antibodypedia</td>
<td>database(Antibodypedia)</td>
<td>xref_antibodypedia</td>
</tr>
<tr>
<td>CPTC</td>
<td>database(CPTC)</td>
<td>xref_cptc</td>
</tr>
<tr>
<td>DNASU</td>
<td>database(DNASU)</td>
<td>xref_dnasu</td>
</tr>
</tbody></table>
<h2 id="genome-annotation-databases">Genome Annotation Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Ensembl</td>
<td>database(Ensembl)</td>
<td>xref_ensembl</td>
</tr>
<tr>
<td>EnsemblBacteria</td>
<td>database(EnsemblBacteria)</td>
<td>xref_ensemblbacteria</td>
</tr>
<tr>
<td>EnsemblFungi</td>
<td>database(EnsemblFungi)</td>
<td>xref_ensemblfungi</td>
</tr>
<tr>
<td>EnsemblMetazoa</td>
<td>database(EnsemblMetazoa)</td>
<td>xref_ensemblmetazoa</td>
</tr>
<tr>
<td>EnsemblPlants</td>
<td>database(EnsemblPlants)</td>
<td>xref_ensemblplants</td>
</tr>
<tr>
<td>EnsemblProtists</td>
<td>database(EnsemblProtists)</td>
<td>xref_ensemblprotists</td>
</tr>
<tr>
<td>GeneID</td>
<td>database(GeneID)</td>
<td>xref_geneid</td>
</tr>
<tr>
<td>Gramene</td>
<td>database(Gramene)</td>
<td>xref_gramene</td>
</tr>
<tr>
<td>KEGG</td>
<td>database(KEGG)</td>
<td>xref_kegg</td>
</tr>
<tr>
<td>MANE-Select</td>
<td>database(MANE-Select)</td>
<td>xref_mane-select</td>
</tr>
<tr>
<td>PATRIC</td>
<td>database(PATRIC)</td>
<td>xref_patric</td>
</tr>
<tr>
<td>UCSC</td>
<td>database(UCSC)</td>
<td>xref_ucsc</td>
</tr>
<tr>
<td>VectorBase</td>
<td>&lt;does not exist&gt;</td>
<td>xref_vectorbase</td>
</tr>
<tr>
<td>WBParaSite</td>
<td>database(WBParaSite)</td>
<td>xref_wbparasite</td>
</tr>
<tr>
<td>WBParaSiteTranscriptProtein</td>
<td>&lt;does not exist&gt;</td>
<td>xref_wbparasitetranscriptprotein</td>
</tr>
</tbody></table>
<h2 id="organism-specific-databases">Organism-Specific Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>ArachnoServer</td>
<td>database(ArachnoServer)</td>
<td>xref_arachnoserver</td>
</tr>
<tr>
<td>Araport</td>
<td>database(Araport)</td>
<td>xref_araport</td>
</tr>
<tr>
<td>CGD</td>
<td>database(CGD)</td>
<td>xref_cgd</td>
</tr>
<tr>
<td>CTD</td>
<td>database(CTD)</td>
<td>xref_ctd</td>
</tr>
<tr>
<td>ConoServer</td>
<td>database(ConoServer)</td>
<td>xref_conoserver</td>
</tr>
<tr>
<td>DisGeNET</td>
<td>database(DisGeNET)</td>
<td>xref_disgenet</td>
</tr>
<tr>
<td>EchoBASE</td>
<td>database(EchoBASE)</td>
<td>xref_echobase</td>
</tr>
<tr>
<td>FlyBase</td>
<td>database(FlyBase)</td>
<td>xref_flybase</td>
</tr>
<tr>
<td>GeneCards</td>
<td>database(GeneCards)</td>
<td>xref_genecards</td>
</tr>
<tr>
<td>GeneReviews</td>
<td>database(GeneReviews)</td>
<td>xref_genereviews</td>
</tr>
<tr>
<td>HGNC</td>
<td>database(HGNC)</td>
<td>xref_hgnc</td>
</tr>
<tr>
<td>HPA</td>
<td>database(HPA)</td>
<td>xref_hpa</td>
</tr>
<tr>
<td>LegioList</td>
<td>database(LegioList)</td>
<td>xref_legiolist</td>
</tr>
<tr>
<td>Leproma</td>
<td>database(Leproma)</td>
<td>xref_leproma</td>
</tr>
<tr>
<td>MGI</td>
<td>database(MGI)</td>
<td>xref_mgi</td>
</tr>
<tr>
<td>MIM</td>
<td>database(MIM)</td>
<td>xref_mim</td>
</tr>
<tr>
<td>MaizeGDB</td>
<td>database(MaizeGDB)</td>
<td>xref_maizegdb</td>
</tr>
<tr>
<td>MalaCards</td>
<td>database(MalaCards)</td>
<td>xref_malacards</td>
</tr>
<tr>
<td>NIAGADS</td>
<td>database(NIAGADS)</td>
<td>xref_niagads</td>
</tr>
<tr>
<td>OpenTargets</td>
<td>database(OpenTargets)</td>
<td>xref_opentargets</td>
</tr>
<tr>
<td>Orphanet</td>
<td>database(Orphanet)</td>
<td>xref_orphanet</td>
</tr>
<tr>
<td>PharmGKB</td>
<td>database(PharmGKB)</td>
<td>xref_pharmgkb</td>
</tr>
<tr>
<td>PomBase</td>
<td>database(PomBase)</td>
<td>xref_pombase</td>
</tr>
<tr>
<td>PseudoCAP</td>
<td>database(PseudoCAP)</td>
<td>xref_pseudocap</td>
</tr>
<tr>
<td>RGD</td>
<td>database(RGD)</td>
<td>xref_rgd</td>
</tr>
<tr>
<td>SGD</td>
<td>database(SGD)</td>
<td>xref_sgd</td>
</tr>
<tr>
<td>TAIR</td>
<td>database(TAIR)</td>
<td>xref_tair</td>
</tr>
<tr>
<td>TubercuList</td>
<td>database(TubercuList)</td>
<td>xref_tuberculist</td>
</tr>
<tr>
<td>VEuPathDB</td>
<td>database(VEuPathDB)</td>
<td>xref_veupathdb</td>
</tr>
<tr>
<td>VGNC</td>
<td>database(VGNC)</td>
<td>xref_vgnc</td>
</tr>
<tr>
<td>WormBase</td>
<td>database(WormBase)</td>
<td>xref_wormbase</td>
</tr>
<tr>
<td>Xenbase</td>
<td>database(Xenbase)</td>
<td>xref_xenbase</td>
</tr>
<tr>
<td>ZFIN</td>
<td>database(ZFIN)</td>
<td>xref_zfin</td>
</tr>
<tr>
<td>dictyBase</td>
<td>database(dictyBase)</td>
<td>xref_dictybase</td>
</tr>
<tr>
<td>euHCVdb</td>
<td>database(euHCVdb)</td>
<td>xref_euhcvdb</td>
</tr>
<tr>
<td>neXtProt</td>
<td>database(neXtProt)</td>
<td>xref_nextprot</td>
</tr>
</tbody></table>
<h2 id="phylogenomic-databases">Phylogenomic Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>GeneTree</td>
<td>database(GeneTree)</td>
<td>xref_genetree</td>
</tr>
<tr>
<td>HOGENOM</td>
<td>database(HOGENOM)</td>
<td>xref_hogenom</td>
</tr>
<tr>
<td>InParanoid</td>
<td>database(InParanoid)</td>
<td>xref_inparanoid</td>
</tr>
<tr>
<td>KO</td>
<td>&lt;does not exist&gt;</td>
<td>xref_ko</td>
</tr>
<tr>
<td>OMA</td>
<td>database(OMA)</td>
<td>xref_oma</td>
</tr>
<tr>
<td>OrthoDB</td>
<td>database(OrthoDB)</td>
<td>xref_orthodb</td>
</tr>
<tr>
<td>PhylomeDB</td>
<td>database(PhylomeDB)</td>
<td>xref_phylomedb</td>
</tr>
<tr>
<td>TreeFam</td>
<td>database(TreeFam)</td>
<td>xref_treefam</td>
</tr>
<tr>
<td>eggNOG</td>
<td>database(eggNOG)</td>
<td>xref_eggnog</td>
</tr>
</tbody></table>
<h2 id="enzyme-and-pathway-databases">Enzyme And Pathway Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>BRENDA</td>
<td>database(BRENDA)</td>
<td>xref_brenda</td>
</tr>
<tr>
<td>BioCyc</td>
<td>database(BioCyc)</td>
<td>xref_biocyc</td>
</tr>
<tr>
<td>PathwayCommons</td>
<td>database(PathwayCommons)</td>
<td>xref_pathwaycommons</td>
</tr>
<tr>
<td>PlantReactome</td>
<td>database(PlantReactome)</td>
<td>xref_plantreactome</td>
</tr>
<tr>
<td>Reactome</td>
<td>database(Reactome)</td>
<td>xref_reactome</td>
</tr>
<tr>
<td>SABIO-RK</td>
<td>database(SABIO-RK)</td>
<td>xref_sabio-rk</td>
</tr>
<tr>
<td>SIGNOR</td>
<td>database(SIGNOR)</td>
<td>xref_signor</td>
</tr>
<tr>
<td>SignaLink</td>
<td>database(SignaLink)</td>
<td>xref_signalink</td>
</tr>
<tr>
<td>UniPathway</td>
<td>database(UniPathway)</td>
<td>xref_unipathway</td>
</tr>
</tbody></table>
<h2 id="miscellaneous-databases">Miscellaneous Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>BioGRID-ORCS</td>
<td>database(BioGRID-ORCS)</td>
<td>xref_biogrid-orcs</td>
</tr>
<tr>
<td>ChiTaRS</td>
<td>database(ChiTaRS)</td>
<td>xref_chitars</td>
</tr>
<tr>
<td>EvolutionaryTrace</td>
<td>database(EvolutionaryTrace)</td>
<td>xref_evolutionarytrace</td>
</tr>
<tr>
<td>GeneWiki</td>
<td>database(GeneWiki)</td>
<td>xref_genewiki</td>
</tr>
<tr>
<td>GenomeRNAi</td>
<td>database(GenomeRNAi)</td>
<td>xref_genomernai</td>
</tr>
<tr>
<td>PHI-base</td>
<td>database(PHI-base)</td>
<td>xref_phi-base</td>
</tr>
<tr>
<td>PRO</td>
<td>database(PRO)</td>
<td>xref_pro</td>
</tr>
<tr>
<td>Pharos</td>
<td>database(Pharos)</td>
<td>xref_pharos</td>
</tr>
<tr>
<td>RNAct</td>
<td>database(RNAct)</td>
<td>xref_rnact</td>
</tr>
</tbody></table>
<h2 id="gene-expression-databases">Gene Expression Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>Bgee</td>
<td>database(Bgee)</td>
<td>xref_bgee</td>
</tr>
<tr>
<td>CleanEx</td>
<td>&lt;does not exist&gt;</td>
<td>xref_cleanex</td>
</tr>
<tr>
<td>CollecTF</td>
<td>database(CollecTF)</td>
<td>xref_collectf</td>
</tr>
<tr>
<td>ExpressionAtlas</td>
<td>database(ExpressionAtlas)</td>
<td>xref_expressionatlas</td>
</tr>
<tr>
<td>Genevisible</td>
<td>database(Genevisible)</td>
<td>xref_genevisible</td>
</tr>
</tbody></table>
<h2 id="family-and-domain-databases">Family And Domain Databases</h2>
<table>
<thead>
<tr>
<th><strong>Label*</strong></th>
<th><strong>Legacy Returned Field</strong></th>
<th><strong>Returned Field</strong></th>
</tr>
</thead>
<tbody><tr>
<td>CDD</td>
<td>database(CDD)</td>
<td>xref_cdd</td>
</tr>
<tr>
<td>DisProt</td>
<td>database(DisProt)</td>
<td>xref_disprot</td>
</tr>
<tr>
<td>Gene3D</td>
<td>database(Gene3D)</td>
<td>xref_gene3d</td>
</tr>
<tr>
<td>HAMAP</td>
<td>database(HAMAP)</td>
<td>xref_hamap</td>
</tr>
<tr>
<td>IDEAL</td>
<td>database(IDEAL)</td>
<td>xref_ideal</td>
</tr>
<tr>
<td>InterPro</td>
<td>database(InterPro)</td>
<td>xref_interpro</td>
</tr>
<tr>
<td>PANTHER</td>
<td>database(PANTHER)</td>
<td>xref_panther</td>
</tr>
<tr>
<td>PIRSF</td>
<td>database(PIRSF)</td>
<td>xref_pirsf</td>
</tr>
<tr>
<td>PRINTS</td>
<td>database(PRINTS)</td>
<td>xref_prints</td>
</tr>
<tr>
<td>PROSITE</td>
<td>database(PROSITE)</td>
<td>xref_prosite</td>
</tr>
<tr>
<td>Pfam</td>
<td>database(Pfam)</td>
<td>xref_pfam</td>
</tr>
<tr>
<td>ProDom</td>
<td>&lt;does not exist&gt;</td>
<td>xref_prodom</td>
</tr>
<tr>
<td>SFLD</td>
<td>database(SFLD)</td>
<td>xref_sfld</td>
</tr>
<tr>
<td>SMART</td>
<td>database(SMART)</td>
<td>xref_smart</td>
</tr>
<tr>
<td>SUPFAM</td>
<td>database(SUPFAM)</td>
<td>xref_supfam</td>
</tr>
<tr>
<td>TIGRFAMs</td>
<td>database(TIGRFAMs)</td>
<td>xref_tigrfams</td>
</tr>
</tbody></table>
</div></div></div></section><div class="VV2Ty"><small> Page last modified: <time datetime="2022-07-19T00:00:00.000Z">Tue Jul 19 2022</time></small></div><h2>Related articles</h2><ul class="data-list no-bullet"><li><section class="card" data-testid="help-card"><div class="card__container"><div class="card__header"><h2 class="small" data-testid="help-title"><a href="https://www.uniprot.org/help/advanced_search">UniProtKB advanced search options</a></h2></div><div class="card__content"></div></div></section></li><li><section class="card" data-testid="help-card"><div class="card__container"><div class="card__header"><h2 class="small" data-testid="help-title"><a href="https://www.uniprot.org/help/customize">Customize display options</a></h2></div><div class="card__content"></div></div></section></li><li><section class="card" data-testid="help-card"><div class="card__container"><div class="card__header"><h2 class="small" data-testid="help-title"><a href="https://www.uniprot.org/help/id_mapping">ID Mapping</a></h2></div><div class="card__content"></div></div></section></li><li><section class="card" data-testid="help-card"><div class="card__container"><div class="card__header"><h2 class="small" data-testid="help-title"><a href="https://www.uniprot.org/help/query-fields">UniProtKB query fields</a></h2></div><div class="card__content"></div></div></section></li><li><section class="card" data-testid="help-card"><div class="card__container"><div class="card__header"><h2 class="small" data-testid="help-title"><a href="https://www.uniprot.org/help/return_fields_databases">UniProtKB return fields for cross-references</a></h2></div><div class="card__content"></div></div></section></li></ul><div class="data-loader__loading"><button class="button secondary" data-testid="click-to-load-more" type="button">Load more articles</button></div></main></div></div><div class="footer"><footer class="OkKEo"><div class="hero-container uniprot-grid uniprot-grid--centered uniprot-grid--with-bleed"><div class="gHijH"><a class="yJxaY" title="UniProt home page" href="https://www.uniprot.org/"><img src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/uniprot-logo.img.0df091.svg" loading="lazy" alt="" width="152" height="70"></a><div class="Yh_HA"><a title="European Bioinformatics Institute" class="external-link" rel="noopener" href="https://www.ebi.ac.uk/" target="_blank"><img src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/embl-ebi-logo.img.48e02e.svg" loading="lazy" alt="" width="231" height="218"></a><a title="Protein Information Resource" class="external-link" rel="noopener" href="https://pir.georgetown.edu/" target="_blank"><img src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/pir-logo.653769.jpg" loading="lazy" alt="" width="231" height="218"></a><a title="Swiss Institute of Bioinformatics" class="external-link" rel="noopener" href="https://www.sib.swiss/" target="_blank"><img src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/sib-logo.d55d59.png" loading="lazy" alt="" width="200" height="152"></a></div></div><div class="BfYvY"><span class="release-info S2LRp zcxGl"><span class="release-info__release_number"><a href="https://www.uniprot.org/help/downloads">Release 2022_03</a></span> | <a href="https://www.uniprot.org/help/release-statistics">Statistics</a></span></div><div class="VcIhJ"><p><a href="https://www.uniprot.org/help/about">Â© 2002 â€“ 2022 UniProt consortium</a></p><p><a href="https://www.uniprot.org/help/license">License &amp; Disclaimer</a> | <a href="https://www.uniprot.org/help/privacy">Privacy Notice</a></p></div><ul class="no-bullet Alml0"><li class="r5vC6"><span class="geMXa">Core data</span><ul class="no-bullet"><li><a href="https://www.uniprot.org/uniprotkb?query=*">Proteins (UniProtKB)</a></li><li><a href="https://www.uniprot.org/proteomes?query=*">Species (Proteomes)</a></li><li><a href="https://www.uniprot.org/uniref?query=*">Protein clusters (UniRef)</a></li><li><a href="https://www.uniprot.org/uniparc?query=*">Sequence archive (UniParc)</a></li></ul></li><li><span class="geMXa">Supporting data</span><ul class="no-bullet"><li><a href="https://www.uniprot.org/citations?query=*">Literature citations</a></li><li><a href="https://www.uniprot.org/taxonomy?query=*">Taxonomy</a></li><li><a href="https://www.uniprot.org/keywords?query=*">Keywords</a></li><li><a href="https://www.uniprot.org/locations?query=*">Subcellular locations</a></li><li><a href="https://www.uniprot.org/database?query=*">Cross-referenced databases</a></li><li><a href="https://www.uniprot.org/diseases?query=*">Diseases</a></li></ul></li><li class="Ljg_Z"><span class="geMXa">Tools</span><ul class="no-bullet"><li><a href="https://www.uniprot.org/blast">BLAST</a></li><li><a href="https://www.uniprot.org/align">Align</a></li><li><a href="https://www.uniprot.org/id-mapping">Retrieve/ID mapping</a></li><li><a href="https://www.uniprot.org/peptide-search">Peptide search</a></li><li><a href="https://www.uniprot.org/tool-dashboard">Tool results</a></li></ul></li><li><span class="geMXa">Information</span><ul class="no-bullet"><li><a href="https://www.uniprot.org/help/publications">Cite <span class="S2LRp">UniProt<sup><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 22" fill="currentColor" width="1em"><path d="M13.65 6.68V7.7a.14.14 0 010 .11c-.13 1.42-.47 2.81-1.01 4.13a15.73 15.73 0 01-2.7 4.54 15.87 15.87 0 01-3.96 3.4c-1.45.86-3 1.53-4.63 1.98 0 0 2.84-4.15 5.8-8.2h-.3a6.82 6.82 0 116.8-6.98zm16.35 0v1.13c-.13 1.42-.48 2.81-1.02 4.13a15.73 15.73 0 01-2.68 4.56 15.86 15.86 0 01-3.96 3.33 18.29 18.29 0 01-4.68 1.99s2.83-4.13 5.79-8.18h-.3A6.82 6.82 0 1130 6.68z"></path></svg></sup></span></a></li><li><a href="https://www.uniprot.org/help/about">About</a> &amp; <a href="https://www.uniprot.org/help">Help</a></li><li><a href="https://www.uniprot.org/help?query=*&amp;facets=category:manual">UniProtKB manual</a></li><li><a href="https://www.uniprot.org/help/technical">Technical corner</a></li><li><a href="https://www.uniprot.org/help?query=*&amp;facets=category:biocuration">Expert biocuration</a></li><li><a href="https://www.uniprot.org/help/release-statistics">Statistics</a></li></ul></li></ul><div class="mD3Hh"><div><p><a href="https://www.uniprot.org/contact">Get in touch <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 13 10" fill="currentColor" width="2ch"><path fill-rule="evenodd" d="M12 1.25C12 .56 11.46 0 10.8 0H1.2C.54 0 0 .56 0 1.25v7.5C0 9.44.54 10 1.2 10h9.6c.66 0 1.2-.56 1.2-1.25v-7.5zm-1.2 0L6 4.38 1.2 1.25h9.6zM6 5.63L1.2 2.5v6.25h9.6V2.5L6 5.63z"></path></svg></a></p><p class="dvAYP"><a title="UniProt posts on Twitter" class="external-link" rel="noopener" href="https://twitter.com/uniprot" target="_blank"><svg viewBox="0 0 24 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg" width="3ch"><path d="M24 2.4c-.9.4-1.8.6-2.8.8A5 5 0 0023.3.4c-1 .5-2 1-3 1.2C19.2.6 18 0 16.5 0a5 5 0 00-4.8 6.2C7.8 6 4.1 4 1.7.9a5.1 5.1 0 001.5 6.8c-.8 0-1.6-.3-2.2-.7v.1a5 5 0 004 5 4.7 4.7 0 01-2.3 0 5 5 0 004.6 3.5A9.7 9.7 0 010 17.7C2.2 19.2 4.8 20 7.5 20c9.1 0 14-7.7 14-14.4V5c1-.7 1.8-1.6 2.5-2.6z"></path></svg></a><a title="UniProt posts on Facebook" class="external-link" rel="noopener" href="https://www.facebook.com/uniprot.org" target="_blank"><svg viewBox="0 0 20 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg" width="3ch"><path d="M10.68 20H1.1C.5 20 0 19.5 0 18.9V1.1C0 .5.5 0 1.1 0h17.8c.6 0 1.1.5 1.1 1.1v17.8c0 .6-.5 1.1-1.1 1.1h-5.1v-7.75h2.6l.39-3.01H13.8V7.3c0-.87.24-1.47 1.5-1.47h1.6v-2.7c-.28-.04-1.23-.12-2.34-.12-2.3 0-3.88 1.4-3.88 4v2.22h-2.6v3.01h2.6V20z"></path></svg></a><a title="UniProt videos on YouTube" class="external-link" rel="noopener" href="https://www.youtube.com/user/uniprotvideos" target="_blank"><svg viewBox="0 0 29 20" xmlns="http://www.w3.org/2000/svg" width="3ch"><g clip-path="url(#youtube-logo_svg__youtube-clip0)"><path d="M11.2 13.7v-8l7.6 4-7.6 4zM28 4.3s-.3-2-1.2-2.8C25.8.4 24.6.4 24 .3 20 0 14.1 0 14.1 0s-6 0-9.9.3c-.5 0-1.7 0-2.8 1.2C.6 2.4.3 4.3.3 4.3S0 6.6 0 9v2.2c0 2.3.3 4.6.3 4.6s.3 2 1.1 2.8c1 1.1 2.5 1.1 3.1 1.2 2.3.2 9.6.3 9.6.3s6 0 9.9-.3c.6 0 1.8 0 2.8-1.2.9-.9 1.2-2.8 1.2-2.8l.2-4.6V8.9L28 4.3z"></path></g><defs><clipPath id="youtube-logo_svg__youtube-clip0"><path d="M0 0h28.2v20H0z"></path></clipPath></defs></svg></a><a title="UniProt blog" class="external-link" rel="noopener" href="https://insideuniprot.blogspot.com/" target="_blank"><svg viewBox="0 0 21 20" xmlns="http://www.w3.org/2000/svg" width="3ch"><g clip-path="url(#blogger-logo_svg__blogger-clip0)"><path d="M13.8 14.8H6.65c-.67 0-1.21-.55-1.21-1.2 0-.66.54-1.2 1.21-1.2h7.15c.66 0 1.2.54 1.2 1.2 0 .65-.54 1.2-1.2 1.2zM6.65 5.17h3.52c.67 0 1.21.54 1.21 1.2 0 .66-.54 1.2-1.21 1.2H6.65c-.67 0-1.21-.54-1.21-1.2 0-.66.54-1.2 1.21-1.2zm13.08 2.57c-.43-.18-2.24.02-2.74-.44-.36-.32-.38-.92-.52-1.7a7.15 7.15 0 00-.57-2.15C15.02 1.59 12.92 0 11 0H6.58A6.35 6.35 0 00.24 6.32v7.37C.24 17.16 3.09 20 6.57 20h7.29a6.35 6.35 0 006.34-6.3l.04-5.11s0-.63-.51-.85z"></path></g><defs><clipPath id="blogger-logo_svg__blogger-clip0"><path transform="translate(.24)" d="M0 0h20v20H0z"></path></clipPath></defs></svg></a><a title="UniProt Google Group" class="external-link" rel="noopener" href="https://groups.google.com/forum/#!forum/ebi-proteins-api" target="_blank"><svg viewBox="0 0 21 20" fill="currentColor" xmlns="http://www.w3.org/2000/svg" width="3ch"><path d="M18.8 0H2.1A1.7 1.7 0 00.5 1.7V20l4.7-3.9h13.6a1.7 1.7 0 001.7-1.7V1.7A1.7 1.7 0 0018.8 0z" fill-opacity="30%"></path><path fill-rule="evenodd" d="M17.1 1.7H3.8a1.7 1.7 0 00-1.7 1.6v9.5a1.7 1.7 0 001.7 1.6h8.3l6.7 5.6V3.3a1.7 1.7 0 00-1.7-1.6zM10.5 6a1.7 1.7 0 11-3.4 0 1.7 1.7 0 013.4 0zM14 8a1.1 1.1 0 10-1.5-1.7A1.1 1.1 0 0014 8zm1.5 2.3s-.6-1.5-2.3-1.5c-.7 0-1.2.3-1.6.6a3.8 3.8 0 00-2.8-1.2c-2.4 0-3.3 1.9-3.3 1.9v1h10v-.8z"></path></svg></a></p></div><p><a class="external-link" rel="noopener" href="https://www.elixir-europe.org/platforms/data/core-data-resources" target="_blank">UniProt is an ELIXIR core data resource<img src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/elixir-cdr.48c606.png" loading="lazy" alt="" width="115" height="115"></a><a title="Core Trust Seal assessment information" class="external-link" rel="noopener" href="https://www.coretrustseal.org/wp-content/uploads/2020/05/UniProt.pdf" target="_blank"><img src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/core-trust-seal-logo.729487.png" loading="lazy" alt="" width="425" height="401"></a></p></div><div class="OjwY2"><span>Main <a href="https://www.uniprot.org/help/about">funding</a> by:</span><a class="external-link" rel="noopener" href="https://www.nih.gov/" target="_blank">National Institutes of Health</a><a title="European Molecular Biology Laboratory" class="external-link" rel="noopener" href="https://www.embl.org/" target="_blank"><img src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/embl-ebi-logo.img.48e02e.svg" loading="lazy" alt="" width="156" height="48"></a><a title="State Secretariat for Education, Research and Innovation SERI" class="external-link" rel="noopener" href="https://www.sbfi.admin.ch/sbfi/en/home.html" target="_blank"><img src="UniProtKB%20return%20fields%20UniProt%20help%20UniProt_files/seri-logo.ff1353.png" loading="lazy" alt="" width="288" height="48"></a></div></div></footer></div></div><div class="gdpr-section">We'd like to inform you that we have updated our <a href="https://www.uniprot.org/help/privacy">Privacy Notice</a> to comply with Europeâ€™s new General Data Protection Regulation (GDPR) that applies since 25 May 2018.<button class="button secondary" type="button">Accept</button></div><span class="NP1Va"><a class="NIoPl Ylh4Z NOeq8" target="_blank" href="https://goo.gl/forms/VrAGbqg2XFg6Mpbh1" rel="noopener noreferrer" tabindex="-1">Feedback</a><a tabindex="-1" class="NIoPl gm7gE" href="https://www.uniprot.org/help">Help</a></span><button class="button primary z6fUI" title="scroll to the top of the page" aria-hidden="true" tabindex="-1" type="button">â¬†</button></div></body></html>